doc/config.rst

```diff
@@ -119,6 +119,53 @@ The following example shows how to use all ``PhpCsFixer`` rules but without the
         ->setFinder($finder)
     ;
 
+If you need to disable or reconfigure a rule for specific files, you can use the ``setRuleCustomizationPolicy`` method:
+
+.. code-block:: php
+
+    <?php
+
+    use PhpCsFixer\Config;
+    use PhpCsFixer\Finder;
+    use PhpCsFixer\Fixer\FixerInterface;
+    use PhpCsFixer\RuleCustomizationPolicyInterface;
+
+    class MyPolicy implements RuleCustomizationPolicyInterface
+    {
+        public function customize(FixerInterface $fixer, SplFileInfo $file): ?FixerInterface
+        {
+            if ('array_syntax' === $fixer->getName()
```

Comment:
*major*

this will fail for proxy fixers, where we operate on new fixer and not the old one anymore - [ref](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/9113) 

my solution ideation can be find here - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/9155 - but it has to be solved for customization/filtering by fixer/path as well.

Reply:
I see the attempt to simplify things, it's good.

I am not sure if i like idea that renamed fixer would lead _crash_ the execution if Policy relies on old name.

Then, having `getRequiredFixerNames` separately means it will have no mechanism ensuring it matches actual rules used in `customize`. 

thoughts ?

Reply:
With that getRequiredFixerNames, the policy will be sure that the expected fixers will be applied.
And if they won't be applied, the execution will abort.

Of course, policy implementations that prefer to ignore fixer replacements can simply have a getRequiredFixerNames method that returns an empty array.
But I'd avoid that: if the policy would like to disable the fixer X, but X has been replaced by Y, I'd prefer to abort the execution instead of having applied an unwanted fixer.

Reply:
with this example:

```php
class MyPolicy implements RuleCustomizationPolicyInterface
{
    public function getRequiredFixerNames(): array {
        return [];
    }

    public function customize(FixerInterface $fixer, \SplFileInfo $file): ?FixerInterface {
        if ($fixer instanceof FooFixer) return null;
        return $fixer;
    }
}
```

how would you ensure that in case of `FooFixer` be replaced by `BarFixer`, the `FooFixer` usage is detected (eg how to ensure it's mentioned in `getRequiredFixerNames`) ?

Reply:
@keradus I understand what you mean, but would you really overcomplicate the policy?

Reply:
Here's what I mean:

https://github.com/mlocati-forks/PHP-CS-Fixer/pull/2/commits/2345b26531e297e883f89ba82afe8a54e8910c56

If that's ok, I can add that commit to this PR #9107